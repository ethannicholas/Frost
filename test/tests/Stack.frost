method main() {
    def s := Stack<Int?>()
    s.push(1)
    s.push(2)
    s.push(3)
    s.push(null)
    s.push(4)
    Console.printLine(s[0])
    Console.printLine(s.pop())
    Console.printLine(s[1])
    Console.printLine(s.pop() !== null)
    Console.printLine(s.pop())
    Console.printLine(s.pop())
    Console.printLine(s.pop())
    Console.printLine(s.count)
}
