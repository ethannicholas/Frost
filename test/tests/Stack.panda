class StackTest {
    @class
    method main() {
        def s := Stack<Int?>()
        s.push(1)
        s.push(2)
        s.push(3)
        s.push(null)
        s.push(4)
        Console.printLine(s.peek())
        Console.printLine(s.pop())
        Console.printLine(s.peek(1))
        Console.printLine(s.pop() != null)
        Console.printLine(s.pop())
        Console.printLine(s.pop())
        Console.printLine(s.pop())
        Console.printLine(s.get_count())
    }
}