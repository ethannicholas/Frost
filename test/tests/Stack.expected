4
4
3
false
3
2
1
0
